packages/zoe/src/zoeService/offer/offer.js

```diff
@@ -9,39 +9,27 @@ import '@agoric/ertp/exported.js';
 import '@agoric/store/exported.js';
 import '../internal-types.js';
 
-const { details: X, quote: q } = assert;
+const { details: X, quote: q, Fail } = assert;
 
-/**
- * @param {Issuer} invitationIssuer
- * @param {GetInstanceAdmin} getInstanceAdmin
- * @param {DepositPayments} depositPayments
- * @param {GetAssetKindByBrand} getAssetKindByBrand
- * @param {GetProposalShapeForInvitation} getProposalShapeForInvitation
- * @returns {Offer}
- */
-export const makeOfferMethod = (
-  invitationIssuer,
-  getInstanceAdmin,
-  depositPayments,
-  getAssetKindByBrand,
-  getProposalShapeForInvitation,
-) => {
+export const makeOfferMethod = offerDataAccess => {
   /** @type {Offer} */
   const offer = async (
     invitation,
     uncleanProposal = harden({}),
     paymentKeywordRecord = harden({}),
     offerArgs = undefined,
   ) => {
+    const invitationIssuer = offerDataAccess.getInvitationIssuer();
     const query = makeInvitationQueryFns(invitationIssuer);
     const { instance, description } = await query.getInvitationDetails(
       invitation,
     );
     // AWAIT ///
 
-    const instanceAdmin = getInstanceAdmin(instance);
+    const instanceAdmin = await offerDataAccess.getInstanceAdmin(instance);
```

Comment:
> AFAICT, it did not previously do an E.when on that argument. Am I missing something?

You are not. When I started responding to your comment on using `callWhen` and `getInstanceAdmin`, I noticed that several of the calls on the `ZoeService` had `instance` parameters that bottomed out in calls in ZoeStorageManager that require the instance to be fulfilled in order to succeed. 

I now think that I put the protection in the wrong place. If the protection is on `ZoeService`, then `ZoeStorageManager` can require that it is called synchronously with a fulfilled `InstanceHandle`, and so the code in `offer` doesn't need an additional `await.  I'll send you a new revision shortly.

Part of my **WDYT** was about the return value. `offer()` used to 
```
   const userSeat = await E(instanceAdmin).makeUserSeat(
      invitationHandle,
      initialAllocation,
      proposal,
      offerArgs,
    );
    // AWAIT ///

    return userSeat;
```
Since there were even more `await`s in that method than before, I removed one by changing that to 
```
    return instanceAdmin.makeUserSeat(
      invitationHandle,
      initialAllocation,
      proposal,
      offerArgs,
    );
```
Since the method is async, there's no difference, right?  Oh, yeah. Previously instanceAdmin was a promise for a local object, but since it's called twice, I `await` it, and then calll it without `E()` both times. So I just wanted you to re-read that method.

Reply:
> Since the method is async, there's no difference, right?

Correct.

Reply:
Is there any remaining uncertainty here, or should we resolve this?